### DISCUSSION TITLE
A spreadsheet to compute the coordinates of the nebula image

### DISCUSSION THREAD

**Discussion by sunshuwei** [Stellarium/stellarium#1556] [Show and tell]
This is a spreadsheet to compute the coordinates of the nebula image.

It only requires four parameters: RA, Dec, Size, Orientation.

![QQ图片20191126213328](https://user-images.githubusercontent.com/34102610/69637965-b5d41200-1094-11ea-93b4-54ebaac15ba6.png)

[Coordinates of DSO Texture.zip](https://github.com/Stellarium/stellarium/files/3892061/Coordinates.of.DSO.Texture.zip)

This is an example.
![0000](https://user-images.githubusercontent.com/34102610/69634585-84a41380-108d-11ea-8658-96cdbfeb27ee.png)

**Comment by alex-w** [comment]
Hmm. Could you write a paragraph for Stellarium User Guide with instructions how to use your spreadsheet for the users?

P.S. We may store your spreadsheets into stellarium-data repository (or as an attachment like DSO catalog) to provide a permanent URL.

@gzotti please comment it.

**Comment by gzotti** [comment]
We have just received a modern tool and docs on how to create the JSON lines for DSO textures. This will go into the next user guide. 
If you can write a complete description on how to use the spreadsheet, we could add another such chapter. Please write it for absolute beginners! E.g. "Image RA hours" etc: Do we need to know the image center, where to get those data etc. How could we proceed when e.g. coordinates of 3 stars in the photo are known, but not the image center? There may be more than one path.

**Comment by gzotti** [comment]
We see some limitations in the .ini format, but were always able to somehow still use it. Moving all configs to JSON may appear to improve some things, for the price of total incompatibility between versions. YAML, yet another format, likely has some benefits, but again for the price of incompatibility. 
Actually, most user interaction is through GUI buttons. What is behind the back should be no user's concern as long as it works. How it works in the background is ugly, indeed, but nobody is expected to edit the oculars.ini in a text editor! 

Unfortunately, our spare time is limited, so we concentrate on things that we think are more urgent than exchanging one "just working" config file format for another "more elegant". But we always accept suggestions in form of working pull requests. And yes, we could use 1-5 more experienced C++/Qt developers, for some topics preferably with a strong astronomical background, to solve deprecation and maybe some efficiency issues.

**Comment by alex-w** [comment]
Alex, sorry, but I see a huge number of problem, who will be introduced by switching to using yaml as config data and zero improvements. If you want to share some parts of ini or all ini file into XML/YAML/JSON/SQL/whatever format or from the XML/YAML/JSON/SQL/whatever format into ini, then just write a script on python/perl/bash/etc. for converting the data. Or define the specification of data files for sharing the oculars/lenses/telescopes/sensor data and write a tool for export/import those data and propose pull request to merge this tool. Is it a big problem? Why you publish you ideas for config data into report, who has nothing common with config data?

**Comment by gzotti** [comment]
Merging .ini files from several users is nowhere part of the regular, documented program use. Users are told that their config files are in their private user data folder, and are their own private data. The more adventurous user will dare to edit them, but it is like hacking the Windows registry: do it carefully. Do some stupid mistake, and your system becomes unusable. 
I hope you understand "Someone Else's Problem". Hacking these .ini files sounds like a very rare use case, given that just editing the Oculars/telescope/etc. data is so simple with the provided GUI, and the format has not changed in the latest versions. If you feel Stellarium should be able to cater for your needs, please develop a working solution. Maybe it is indeed easier to write simple converters .ini->YAML, edit/merge/whatever the YAML, then convert back YAML->.ini.

**Comment by axd1967** [comment]
update: YAML (.yml) format is now appearing in Stellarium `master` (even if for build use only).

**Comment by alex-w** [comment]
No YAML for usage within Stellarium!

**Comment by axd1967** [comment]
(See also https://github.com/Stellarium/stellarium-data/tree/master/adding-nebula-images)
